lib/libc/misc/Make.defs

```diff
@@ -69,7 +69,7 @@ CSRCS += lib_ioctl.c
 endif
 
 else
-ifneq ($(CONFIG_NSOCKET_DESCRIPTORS),0)
```

Comment:
I mean ifneq ($(CONFIG_NSOCKET_DESCRIPTORS),0) does not work properly.
I'm not sure why this does not work properly. 
The variable which is set to automatically, should not be used in Make.defs for conditional compile.